[VP8/VP9] Add support for alternate codec name

youssefsoli opened the pull request:
#### eccf68f1d1e90ed5e3ed8d2d04176f54b11f3015
<pre>
[VP8/VP9] Add support for alternate codec name
<a href="https://bugs.webkit.org/show_bug.cgi?id=242790">https://bugs.webkit.org/show_bug.cgi?id=242790</a>

Reviewed by Jean-Yves Avenard.

Currently the VPx codec parser in both VP9Utilities and SourceBufferParserWebM
do not recognize the alternate form of no parameter codec names (vp9.0 and vp8.0).

From the spec: `Note that, currently, the attribute string value &quot;vp8&quot; may
also be expressed as &quot;vp8.0&quot;, and &quot;vp9&quot; may be expressed as &quot;vp9.0&quot;.`

* Source/WebCore/platform/graphics/VP9Utilities.cpp:
(WebCore::parseVPCodecParameters):
* Source/WebCore/platform/graphics/cocoa/SourceBufferParserWebM.cpp:
(WebCore::SourceBufferParserWebM::isContentTypeSupported):

* LayoutTests/media/vp-codec-parameters-expected.txt:
* LayoutTests/media/vp-codec-parameters.html:

Canonical link: <a href="https://commits.webkit.org/252535@main">https://commits.webkit.org/252535@main</a>
</pre>

jyavenard reviewed (changes requested):
This needs tests.

There's a an Internal test checking parseVPCodecParamenters.

Adding something like:

```
        testExpectedVPCodecConfigurationRecord('internals.parseVPCodecParameters("vp9")', makeVPCodecConfigurationRecord('vp09', 0, 10, 8, 1, 1, 1, 1, 0));
        testExpectedVPCodecConfigurationRecord('internals.parseVPCodecParameters("vp8")', makeVPCodecConfigurationRecord('vp08', 0, 10, 8, 1, 1, 1, 1, 0));
        testExpectedVPCodecConfigurationRecord('internals.parseVPCodecParameters("vp9.0")', makeVPCodecConfigurationRecord('vp09', 0, 10, 8, 1, 1, 1, 1, 0));
        testExpectedVPCodecConfigurationRecord('internals.parseVPCodecParameters("vp8.0")', makeVPCodecConfigurationRecord('vp08', 0, 10, 8, 1, 1, 1, 1, 0));
        consoleWrite('Test invalid number of required parameters:')
        testExpected('internals.parseVPCodecParameters("vp09")', null);
        testExpected('internals.parseVPCodecParameters("vp09.00")', null);
        testExpected('internals.parseVPCodecParameters("vp09.00.41")', null);
        testExpected('internals.parseVPCodecParameters("vp9.0.")', null);
        testExpected('internals.parseVPCodecParameters("vp8.0.")', null);
```

youssefsoli committed:
[VP8/VP9] Add support for alternate codec name
https://bugs.webkit.org/show_bug.cgi?id=242790

Reviewed by Jean-Yves Avenard.

Currently the VPx codec parser in both VP9Utilities and SourceBufferParserWebM
do not recognize the alternate form of no parameter codec names (vp9.0 and vp8.0).

From the spec: `Note that, currently, the attribute string value "vp8" may
also be expressed as "vp8.0", and "vp9" may be expressed as "vp9.0".`

* Source/WebCore/platform/graphics/VP9Utilities.cpp:
(WebCore::parseVPCodecParameters):
* Source/WebCore/platform/graphics/cocoa/SourceBufferParserWebM.cpp:
(WebCore::SourceBufferParserWebM::isContentTypeSupported):

* LayoutTests/media/vp-codec-parameters-expected.txt:
* LayoutTests/media/vp-codec-parameters.html:

Canonical link: https://commits.webkit.org/252535@main
